recommendation/widedeep/widedeep.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2022, Ampere Computing LLC
+
+import bz2
+import math
+import pickle
+import collections
+
+import numpy as np
+import tensorflow as tf
+
+import utils.misc as utils
+
+
+class WideDeep:
+    """
+    A class providing facilities for preprocessing and postprocessing of ImageNet validation dataset.
+    """
+
+    def __init__(self, batch_size: int, config=None, runner=None, dataset_path=None, tfrecords_path=None):
+
+        if tfrecords_path is None:
+            env_var = "WIDEDEEP_TFRECORDS_PATH"
+            tfrecords_path = utils.get_env_variable(
+                env_var, f"Path to tfrecords path has not been specified with {env_var} flag")
+            self.tfrecords_path = tfrecords_path
+
+        self.available_instances = sum(1 for _ in tf.compat.v1.python_io.tf_record_iterator(
+            self.tfrecords_path))
+
+        self.dataset_path = dataset_path
+        self.batch_size = batch_size
+        self.no_of_batches = math.ceil(float(self.available_instances // self.batch_size))
```

Comment:
what's this math ceil and float for? // operator already delivers you a rounded-down int